src/server/database/Updater/DBUpdater.cpp

```diff
@@ -40,10 +40,10 @@ std::string DBUpdaterUtil::GetCorrectedMySQLExecutable()
 bool DBUpdaterUtil::CheckExecutable()
 {
     boost::filesystem::path exe(GetCorrectedMySQLExecutable());
-    if (!exists(exe))
+    if (is_directory(exe))
```

Comment:
shouldnt this be !is_regular_file?

Reply:
Hmm, I think we should even do something like
```c++
if (is_regular_file(exe))
    return true;
if (is_directory(exe))
{
    ...
}
return false;
```
With that we would be able to catch the `is_block_file`, `is_character_file`, etc. cases.

Reply:
I think `is_regular_file` would be the best choice here too.

Reply:
Im pretty sure we only care that is_regular_file(exe) -> good
else search in path ignoring config

possibly maybe is_symlink? (but then you need to check link target) nothing else should be treated as valid entry

Reply:
Argh, sorry. I misinterpreted `SearchExecutableInPath`

In this case I agree, let's do `!is_regular_file`